main.go

```diff
@@ -80,8 +80,8 @@ func main() {
 		opts.Usage()
 		os.Exit(0)
 	}
-
-	storeBuilder := store.NewBuilder(ctx)
+	ksmMetricsRegistry := prometheus.NewRegistry()
+	storeBuilder := store.NewBuilder(ctx, ksmMetricsRegistry)
```

Comment:
for the better or worse, this should continue to use the same builder pattern I think, at least for consistency if nothing else :)

Reply:
meaning, to change this to `WithRegistry(ksmMetricsRegistry)`?

Reply:
@LiliC Is it possible to follow the same pattern as we do currently for this?

```
func (b *Builder) buildDaemonSetStore() *metricsstore.MetricsStore {
	return b.buildStore(daemonSetMetricFamilies, &appsv1.DaemonSet{}, createDaemonSetListWatch)
}

func (b *Builder) buildDeploymentStore() *metricsstore.MetricsStore {
	return b.buildStore(deploymentMetricFamilies, &appsv1.Deployment{}, createDeploymentListWatch)
}

func (b *Builder) buildEndpointsStore() *metricsstore.MetricsStore {
	return b.buildStore(endpointMetricFamilies, &v1.Endpoints{}, createEndpointsListWatch)
}
```
So that we pass the method that returns the new ListerWatcher impl as a function param instead of injecting it as a new struct field param. I prefer not changing the struct signature of `whiteBlackLister` for the sake of restricting its scope to only white & black listing.

Reply:
@tariq1890 we are changing the `Builder` struct, not the `whiteBlackLister` struct, just the github diff displays it silly.

Reply:
Whoops, sorry about that.